AAUnlimited/Functions/AAPlay/Subs.cpp

```diff
@@ -1,33 +1,147 @@
 #include "StdAfx.h"
 
 #include <list>
+#include <regex>
 
 namespace Subtitles {
 
-	std::wstring text;
-	std::list<std::wstring> lines;
-	RECT rect = { 0, 40, 1024, 256 };
-	D3DCOLOR color = D3DCOLOR_RGBA(255, 255, 255, 255);
+	std::list<std::tuple< std::wstring, int >> lines;
+	const int fontLayersNum = 9;
```

Comment:
This isn't a good approach to do outlines. `DrawText` is an expensive function and we can't call it 10 times per line every frame. At most you should do it once to make a parallel shadow effect instead of outlines.

Reply:
answered in main comment